### PR TITLE
Consider the case of an empty Collection in parameters

### PR DESCRIPTION
**Pull request by ChristianZhou**

### CODE REVIEW

File: core/mybatis-generator-core/src/main/java/org/mybatis/generator/codegen/mybatis3/model/ExampleGenerator.java [hunk 1]
```diff
@@ -243,9 +243,15 @@ private InnerClass getCriterionInnerClass() {
         method.addBodyLine("this.condition = condition;"); //$NON-NLS-1$
         method.addBodyLine("this.value = value;"); //$NON-NLS-1$
         method.addBodyLine("this.typeHandler = typeHandler;"); //$NON-NLS-1$
-        method.addBodyLine("if (value instanceof List<?>) {"); //$NON-NLS-1$
+        method.addBodyLine("if (value instanceof Collection<?>) {"); //$NON-NLS-1$
+        method.addBodyLine("if (((Collection<?>) value).size() != 0) {"); //$NON-NLS-1$
         method.addBodyLine("this.listValue = true;"); //$NON-NLS-1$
         method.addBodyLine("} else {"); //$NON-NLS-1$
+        method.addBodyLine("this.noValue = true;"); //$NON-NLS-1$
+        method.addBodyLine("if (value instanceof Collection<?>) {"); //$NON-NLS-1$
```
Comment (ChristianZhou) [inline core/mybatis-generator-core/src/main/java/org/mybatis/generator/codegen/mybatis3/model/ExampleGenerator.java:251]: Delete this line

File: core/mybatis-generator-core/src/main/java/org/mybatis/generator/codegen/mybatis3/model/ExampleGenerator.java [hunk 2]
```diff
@@ -243,9 +243,14 @@ private InnerClass getCriterionInnerClass() {
         method.addBodyLine("this.condition = condition;"); //$NON-NLS-1$
         method.addBodyLine("this.value = value;"); //$NON-NLS-1$
         method.addBodyLine("this.typeHandler = typeHandler;"); //$NON-NLS-1$
-        method.addBodyLine("if (value instanceof List<?>) {"); //$NON-NLS-1$
+        method.addBodyLine("if (value instanceof Collection<?>) {"); //$NON-NLS-1$
```
Comment (jeffgbutler) [inline core/mybatis-generator-core/src/main/java/org/mybatis/generator/codegen/mybatis3/model/ExampleGenerator.java:246]: This should be List - not Collection. The method signature that calls this is List.

File: core/mybatis-generator-core/src/main/java/org/mybatis/generator/codegen/mybatis3/model/ExampleGenerator.java [hunk 3]
```diff
@@ -243,9 +243,14 @@ private InnerClass getCriterionInnerClass() {
...
+        method.addBodyLine("if (((Collection<?>) value).size() != 0) {"); //$NON-NLS-1$
```
Comment (jeffgbutler) [inline core/mybatis-generator-core/src/main/java/org/mybatis/generator/codegen/mybatis3/model/ExampleGenerator.java:247]: Same comment - List not Collection

**Review by jeffgbutler — COMMENTED** [review]
Please add tests to show the intended results of this.

File: core/mybatis-generator-core/src/main/java/org/mybatis/generator/codegen/mybatis3/model/ExampleGenerator.java (hunk 2 above)
Comment (ChristianZhou) [reply core/mybatis-generator-core/src/main/java/org/mybatis/generator/codegen/mybatis3/model/ExampleGenerator.java:246]: image
It's an Object，I can use Collection like HashSet as well. By the way, what's more important is that  an empty Collection in parameters occurs exception like "java.lang.reflect.InvocationTargetException"

**Comment by jeffgbutler** [comment]
This breaks the build, has no tests, and no documentation.

**Pull request closed without merging**